File: stage2/s2_for_03.js
// S2_FOR_03 — Sum until threshold
function sumUntil(nums, threshold) {
  let sum = 0;

  for (const n of nums) {
    if (sum + n > threshold) {
      break;
    }
    sum += n;
  }

  return sum;
}

// Tests
console.log(sumUntil([1, 2, 3, 4, 5], 7)); // 6  (1+2+3)
console.log(sumUntil([5, 5, 5], 9)); // 5
console.log(sumUntil([10, 20], 5)); // 0

File: stage2/B/S2_FOR_03.py
# S2_FOR_03.py

def sum_until(nums, threshold):
    total = 0
    for n in nums:
        if total + n > threshold:
            break
        total += n
    return total


def main():
    tests = [
        ([1, 2, 3, 4], 6),
        ([5, 5, 5], 9),
        ([10, -3, 2], 7),
        ([], 10),
    ]
    for nums, th in tests:
        print(f"nums={nums!r}, threshold={th} -> {sum_until(nums, th)!r}")


if __name__ == "__main__":
    main()
